classes/migration/upgrade/v3_4_0/PreflightCheckMigration.php

```diff
@@ -205,7 +205,10 @@ protected function checkUsageStatsLogs(): void
         $usageStatsDir = StatisticsHelper::getUsageStatsDirPath();
         // check if there are usage stats log files older than yesterday
         foreach (glob($usageStatsDir . '/usageEventLogs/*') as $usageStatsLogFile) {
-            $lastModified = date('Ymd', filemtime($usageStatsLogFile));
+            if (!preg_match('/\d{8}\.log$/', $usageStatsLogFile, $lastModified)) {
+                throw new Exception("The log file \"{$usageStatsLogFile}\" doesn't follow the expected naming pattern \"usage_events_YearMonthDay.log\"");
+            }
+            [$lastModified] = $lastModified;
```

Comment:
I think you need to use `'/(\d{8})\.log$/'` in preg_match above, and then `$lastModified = $lastModified[1]` right?

Reply:
Ops, I tagged you before reviewing and creating the PR for the tests 🙈
I'll change the name too 👌